### PR TITLE
fix(ESSNTL-4746): LastSeen filter backend edge case

### PR DESCRIPTION
**Pull request by adonispuente**
The backend returns a 503 in specific scenarios, this PR adds additional edge cases to lastSeen filter while the backend figures out the issue. This also allows the check on both ends. 

To test, 
Try entering random strings in the custom date inputs, 
Try entering incomplete dates, like 201 or 192,
Enter years well past our time like 1900s

### CODE REVIEW

File: src/components/filters/useLastSeenFilter.js [hunk 1]
```diff
@@ -93,7 +93,7 @@ export const useLastSeenFilter = (
         const todaysDate = moment().endOf('day');
         const selectedFromDate = moment(date).startOf('day');
 
-        if (!containsSpecialChars(date) && selectedFromDate < todaysDate) {
+        if (!containsSpecialChars(date) && selectedFromDate < todaysDate && date.length > 9 && selectedFromDate > oldestDate) {
```
Comment (bastilian) [inline src/components/filters/useLastSeenFilter.js:96]: Is there something that dictates what the `date.length` has to be, so we don't need to use a hardcoded `9` here?

Comment (adonispuente) [reply src/components/filters/useLastSeenFilter.js:96]: its more like a 'standard; length because with the format used in the component, MM-DD-YYYY, you have to have at least 10 characters in the field, similar to a phone number

Comment (bastilian) [reply src/components/filters/useLastSeenFilter.js:96]: I think in this case we could have it as a descriptive constant like `DEFAULT_DATE_LENGTH`.

**Comment by adonispuente** [comment]
@bastilian pinging this back up

**Review by gkarat — APPROVED** [review]
LGTM, tried locally the scenarios you provided in the desc. Overall , I have a feeling we could have just wrapped the place in the code where it makes a request with try-catch block. so every time something is invalid, we will still request server and then if it's 503, write that the date is invalid

**Comment by adonispuente** [comment]
@gkarat I like this idea. Right now its fired off in api.js , in a promise. It would cause a rewrite of how that request is sent but it does save us from having to implement more failsafes in the future. I will look into this

**Comment by bastilian** [comment]
@adonispuente Thank you for changing that constant otherwise looks good to me. I'll leave the final approval up to @gkarat. :)

**Review by bastilian — APPROVED** [review]

**Comment by adonispuente** [comment]
The minimum date the backend allowed changed so ive updated it. There was an issue when deleting dates with this new const so I added the edge case for that as well

File: src/components/filters/helpers.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import moment from 'moment';
 
-export const oldestDate = new Date(1950, 1, 1);
+export const oldestDate = new Date(1970, 1, 1);
```
Comment (bastilian) [inline src/components/filters/helpers.js:3]: btw.: Screenshot 2023-06-27 at 00 44 47

Comment (bastilian) [reply src/components/filters/helpers.js:3]: You can pass Date a value when constructing. From the MDN docs[0]:

> An integer value representing the [timestamp](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#the_epoch_timestamps_and_invalid_date) (the number of milliseconds since midnight at the beginning of January 1, 1970, UTC — a.k.a. the [epoch](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#the_epoch_timestamps_and_invalid_date)).


So, O milliseconds since January 1, 1970, gives January 1, 1970. :D

[0] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date

Comment (adonispuente) [reply src/components/filters/helpers.js:3]: That doesnt seem like new. Date(0), this is the oldestDate const I updated

Comment (adonispuente) [reply src/components/filters/helpers.js:3]: Oh thats actually just a huge coincidence haha

Comment (bastilian) [reply src/components/filters/helpers.js:3]: It's not a coincidence: https://en.wikipedia.org/wiki/Unix_time

But yeah, it does make sense to have it explicit. Maybe. :)

Comment (bastilian) [reply src/components/filters/helpers.js:3]: That's probably also why the backend has that as the minimum date. :)

**Comment by gkarat** [comment]
Gonna re-run the Travis Test step

**Comment by gkarat** [comment]
LGTM
